blockchain/chain_sync/src/network_context.rs

```diff
@@ -369,18 +362,13 @@ where
                 response_channel: tx,
             })
             .await
-            .map_err(|_| "Failed to send hello request: receiver dropped")?;
+            .map_err(|_| anyhow::anyhow!("Failed to send hello request: receiver dropped"))?;
```

Comment:
You can make use of context here. See https://docs.rs/anyhow/latest/anyhow/trait.Context.html

Reply:
Good one! I didn't know that, fixed